miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/ExperimentRestControllerST.java

```diff
@@ -0,0 +1,141 @@
+package uk.ac.bbsrc.tgac.miso.webapp.springtest;
+
+import org.junit.Test;
+
+import org.springframework.web.servlet.*;
+
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.*;
+
+
+import org.springframework.security.test.context.support.WithMockUser;
+import static org.junit.Assert.*;
+import java.util.List;
+import java.util.ArrayList;
+
+import javax.ws.rs.core.MediaType;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.*;
+import com.jayway.jsonpath.JsonPath;
+
+import static org.hamcrest.Matchers.*;
+
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.core.data.Experiment;
+import uk.ac.bbsrc.tgac.miso.core.data.InstrumentModel;
+import uk.ac.bbsrc.tgac.miso.core.data.KitImpl;
+import uk.ac.bbsrc.tgac.miso.core.data.Experiment.RunPartition;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.LibraryImpl;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.StudyImpl;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.kit.KitDescriptor;
+import uk.ac.bbsrc.tgac.miso.dto.ExperimentDto;
+import uk.ac.bbsrc.tgac.miso.dto.KitConsumableDto;
+import uk.ac.bbsrc.tgac.miso.dto.ExperimentDto.RunPartitionDto;
+import uk.ac.bbsrc.tgac.miso.core.data.Run;
+
+
+
+public class ExperimentRestControllerST extends AbstractST {
+
+  private static final String CONTROLLER_BASE = "/rest/experiments";
+  private static final Class<Experiment> entityClass = Experiment.class;
+
+
+  @Test
+  public void testAddKit() throws Exception {
+    KitImpl kit = currentSession().get(KitImpl.class, 1);
+
+    KitConsumableDto dto = Dtos.asDto(kit);
+    getMockMvc()
+        .perform(post(CONTROLLER_BASE + "/2/addkit").content(makeJson(dto)).contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk());
+
+    Experiment added = currentSession().get(entityClass, 2);
+    assertEquals(1, added.getKits().size());
+    assertEquals(kit.getId(), added.getKits().iterator().next().getId());
+    assertEquals(kit.getName(), added.getKits().iterator().next().getName());
+    // just checking that the right kit was added to the experiment
+  }
+
+  @Test
+  public void testAddPartition() throws Exception {
+    getMockMvc().perform(post(CONTROLLER_BASE + "/2/add").param("runId", "1").param("partitionId", "11"))
+        .andExpect(status().isOk());
+
+    Experiment exp2 = currentSession().get(entityClass, 2);
+    assertEquals(exp2.getRunPartitions().get(0).getRun(), currentSession().get(Run.class, 1));
```

Comment:
* Expected and actual are backwards
* Test the run ID rather than the Run itself, which would depend on `Run#equals`
* Check the actual partition too since that's what is being added